src/EachPromise.php

```diff
@@ -96,6 +96,22 @@ public function promise(): PromiseInterface
 
     private function createPromise(): void
     {
+        // Clear the references when the promise is resolved.
+        $clearFn = function (): void {
+            $this->iterable = $this->concurrency = $this->pending = null;
+            $this->onFulfilled = $this->onRejected = null;
+            $this->nextPendingIndex = 0;
+        };
+
+        // In the case of empty, create a promise that will immediately become resolved via wait() or
+        // via Utils::queue()->run() (https://github.com/guzzle/promises/issues/176). We could simply
+        // create a fulfilled promise here, but that would be an observable behavioral change (see
+        // EachPromiseTest::testResolvesInCaseOfAnEmptyListAndInvokesFulfilled)
```

Comment:
Curious if you agree with the approach taken here vs. changing testResolvesInCaseOfAnEmptyListAndInvokesFulfilled to call `Utils::queue()->run()` before checking that fulfilled is called.

Originally I had thought it would make sense for an empty EachPromise to resolve synchronously without even needing to pump the queue.

However, this solution works just as well for me in terms of making EachPromise "fiber-friendly" and preserves the existing behavior more exactly.